check-dependencies.sh

```diff
@@ -0,0 +1,161 @@
+#!/bin/bash
+set -euo pipefail
+
+# Testing with this to simulate different installed apps:
+# docker run -it -v ~/Source/aws-cdk:/var/cdk ubuntu bash
+
+# Note: Don't use \d in regex, it doesn't work on Macs without GNU tools installed
+# Use [0-9] instead
+
+app=""
+app_min=""
+app_v=""
+
+die() { echo "$*" 1>&2 ; exit 1; }
+wrong_version() { echo "Found $app version $app_v. Install $app >= $app_min" 1>&2; exit 1; }
+
+check_which() {
+    app=$1
+    min=$2
+
+    echo -e "Checking if $app is installed... \c"
+
+    w=$(which ${app}) || w=""
+
+    if [ -z $w ] || [ $w == "$app not found" ]
+    then
+        die "Missing dependency: $app. Install $app >= $min"
+    else
+        echo "Ok"
+    fi
+}
+
+# [Node.js >= 10.13.0]
+#   ⚠️ versions `13.0.0` to `13.6.0` are not supported due to compatibility issues with our dependencies.
+app="node"
+app_min="v10.13.0"
+check_which $app $app_min
+app_v=$(node --version)
+
+# Check for version 10.*.* - 29.*.*
+echo -e "Checking node version... \c"
+if [ $(echo $app_v | grep -c -E "v[12][0-9]\.[0-9]+\.[0-9]+") -eq 1 ]
+then    
+    # Check for version 13.0 to 13.6
+    if [ $(echo $app_v | grep -c -E "v13\.[0-6]\.[0-9]+") -eq 1 ]
+    then
+        die "node versions 13.0.0 to 13.6.0 are not supported due to compatibility issues with our dependencies."
+    else
+        # Check for version < 10.13
+        if [ $(echo $app_v | grep -c -E "v10\.([0-9]|1[0-2])\.[0-9]+") -eq 1 ]
+        then   
+            wrong_version
+        else
+            echo "Ok"
+        fi
+    fi
+else    
+    echo "Not 12"
+    wrong_version
+fi
+
+# TODO - Should we switch to JS here instead of bash since we know node is installed?
```

Comment:
bash is awesome

Reply:
I agree with Romain that we are making it harder to eventually modify the build to be friendly to Windows developers. The more we do in TS/JS, the less duplicate code we have to write in Powershell.